### PR TITLE
packages/microsoft_sqlserver: Improve documentation

### PR DESCRIPTION
**Pull request by shmsr**
## Proposed commit message

Improve documentation to elaborate more on permissions. This PR also includes some general improvements to the documentation.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Related issues

- https://github.com/elastic/integrations/issues/11483

### CODE REVIEW

File: packages/microsoft_sqlserver/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.9.3"
+  changes:
+    - description: Improve documentation to add more information about required permissions.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/99999
```
Comment (andrewkroh) [inline packages/microsoft_sqlserver/changelog.yml:6]:
```suggestion
      link: https://github.com/elastic/integrations/pull/11646
```

File: packages/microsoft_sqlserver/_dev/build/docs/README.md [hunk 2]
```diff
@@ -42,6 +40,19 @@ If you browse Microsoft Developer Network (MSDN) for the following tables, you w
...
     - [sys.dm_os_performance_counters](https://learn.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-os-performance-counters-transact-sql?view=sql-server-ver16)
 
+Please make sure the user has the permissions to system as well as user-defined databases. For the particular user used in the integration, the following requirements are met:
+
+User setup options:
+
+- Grant specific permissions as mentioned in the MSDN pages above.
+- Alteratively, use `sysadmin` role (includes all required permissions): This can be configured via SQL Server Management Studio (SSMS) in `Server Roles`.
+
+User Mappings (using SQL Server Management Studio (SSMS)):
```
Comment (muthu-mps) [inline packages/microsoft_sqlserver/_dev/build/docs/README.md:50]: As this is more of how do we assign a role to the user in SQL server. Can we add the reference link from the official documentation?
Documentation [link](https://learn.microsoft.com/en-us/sql/relational-databases/security/authentication-access/join-a-role?view=sql-server-ver16#SSMSProcedure:~:text=in%20db_securityadmin.-,Using%20SQL%20Server%20Management%20Studio,-Note).

Comment (shmsr) [reply packages/microsoft_sqlserver/_dev/build/docs/README.md:50]: Done!

**Review by muthu-mps — APPROVED** [review]
LGTM!

**Pull request merged**